### PR TITLE
Fix multistore

### PR DESCRIPTION
**Pull request by alvinjohnsonso**
Changes made to fix multistore support (Both for Prestashop 1.6 and 1.7):
1. `controllers/admin/AdminTawktoController.php`:
    - Removed `domain`, `display_opts`, `page_id`, `widget_id`, and `same_user` in template data as these would be retrieved by the client later on.
    - Added private function `getStoreDetails` function and returned the results in template data as shops. Contains store's id, name, and domain.
    - Added controller function `displayAjaxGetStoreWidget`. This will be used by client to retrieve the selected store's widget config(`page_id`, `widget_id`, and `same_user`).
    - Added controller function `displayAjaxGetStoreVisibilityOpts`. This will be used by the client to retrieve the selected store's visibility options.
    - Changed `Configuration::updateValue` to `Configuration::updateGlobalValue` to enable updating of other stores even if the admin user changed into another shop group.

2. `views/templates/admin/tawkto/helpers/view/view.tpl`:
    - Transferred the scripts to the bottom of the file.
    - Added a section for the store selector field.
    - Removed smarty(tpl) syntax for behavior and data binding of visibility options, visibility warning, and widget already set warning. Instead used JQuery to handle these. The reason for this is that smarty(tpl) is a server-side template injection and doesn't have client-side data bindings to update the data bound to the elements.
    - Added `getStoreWidget` function to retrieve store widget config in the server through the controller function `displayAjaxGetStoreWidget` using ajax.
    - Added `getStoreVisibilityOpts` function to retrieve store visibility options in the server through the controller function `displayAjaxGetStoreVisibilityOpts` using ajax.
    - Added an alert error element for `getStoreWidget` and `getStoreVisibilityOpts` to receive an error.

3. `tawkto.php` (only for Prestashop 1.6)
    - updated logic on extracting slug listed on visibility options.

4. UI Updates:
    - Prestashop 1.6
        - prestashop-1 6

    - Prestashop 1.7
        - prestashop-1 7

### CODE REVIEW

File: prestashop1.6/views/templates/admin/tawkto/helpers/view/view.tpl [hunk 1]
```diff
@@ -15,123 +15,81 @@
...
-{/if}
-
-<iframe
-    id="tawkIframe"
-    src=""
-    style="min-height: 275px; width: 100%; border: none; margin: 5px 0; padding: 10px; background: #FFF;">
-</iframe>
-<input type="hidden" class="hidden" name="page_id" value="{$page_id|escape:'html':'UTF-8'}">
-<input type="hidden" class="hidden" name="widget_id" value="{$widget_id|escape:'html':'UTF-8'}">
-<script type="text/javascript">
```
Comment (alvinjohnsonso) [inline prestashop1.6/views/templates/admin/tawkto/helpers/view/view.tpl:32]: This script block was just moved below in [line 170](https://github.com/tawk/tawk-prestashop/pull/17/files#diff-2f067937ab3c16db22129b4ff143c0d6dda6b33cb00e14a8a60ee6e4851c34d2R170)

File: prestashop1.6/views/templates/admin/tawkto/helpers/view/view.tpl [hunk 2]
```diff
@@ -15,123 +15,81 @@
...
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  *}
-
 {include file="toolbar.tpl" toolbar_btn=$toolbar_btn toolbar_scroll=$toolbar_scroll title=$title}
 
-{if !$same_user}
-<div id="widget_already_set" style="width: 50%; float: left; color: #3c763d; border-color: #d6e9c6; font-weight: bold;" class="alert alert-warning">Widget set by other user</div>
-{/if}
-
-<iframe
```
Comment (alvinjohnsonso) [inline prestashop1.6/views/templates/admin/tawkto/helpers/view/view.tpl:25]: This iframe was moved in [line 44](https://github.com/tawk/tawk-prestashop/pull/17/files#diff-2f067937ab3c16db22129b4ff143c0d6dda6b33cb00e14a8a60ee6e4851c34d2R44)

File: prestashop1.7/views/templates/admin/tawkto/helpers/view/view.tpl [hunk 3]
```diff
@@ -15,108 +15,45 @@
...
-{/if}
-
-<iframe
-    id="tawkIframe"
-    src=""
-    style="min-height: 275px; width: 100%; border: none; margin: 5px 0; padding: 10px; background: #FFF;">
-</iframe>
-<input type="hidden" class="hidden" name="page_id" value="{$page_id|escape:'html':'UTF-8'}">
-<input type="hidden" class="hidden" name="widget_id" value="{$widget_id|escape:'html':'UTF-8'}">
-<script>
```
Comment (alvinjohnsonso) [inline prestashop1.7/views/templates/admin/tawkto/helpers/view/view.tpl:32]: This script block was move to [line 170](https://github.com/tawk/tawk-prestashop/pull/17/files#diff-c02860703bdfe1339eaa7472ceb72f0a6cec22f9302c1f1ba3042a8b81a9b920R170)

File: prestashop1.7/views/templates/admin/tawkto/helpers/view/view.tpl [hunk 4]
```diff
@@ -15,108 +15,45 @@
...
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  *}
-
 {include file="toolbar.tpl" toolbar_btn=$toolbar_btn toolbar_scroll=$toolbar_scroll title=$title}
 
-{if !$same_user}
-<div id="widget_already_set" style="width: 50%; float: left; color: #3c763d; border-color: #d6e9c6; font-weight: bold;" class="alert alert-warning">Widget set by other user</div>
-{/if}
-
-<iframe
```
Comment (alvinjohnsonso) [inline prestashop1.7/views/templates/admin/tawkto/helpers/view/view.tpl:25]: This iframe was moved to [line 44](https://github.com/tawk/tawk-prestashop/pull/17/files#diff-c02860703bdfe1339eaa7472ceb72f0a6cec22f9302c1f1ba3042a8b81a9b920R44)

File: prestashop1.6/controllers/admin/AdminTawktoController.php [hunk 5]
```diff
@@ -56,50 +57,13 @@ public function initToolbar()
...
-            'display_opts' => $displayOpts,
-            'page_id' => Configuration::get(TawkTo::TAWKTO_WIDGET_PAGE_ID."_{$shopId}"),
-            'widget_id' => Configuration::get(TawkTo::TAWKTO_WIDGET_WIDGET_ID."_{$shopId}"),
-            'same_user' => $sameUser
-        );
+                'iframe_url' => $this->getIframeUrl(),
+                'base_url'   => $this->getBaseUrl(),
+                'controller' => $this->context->link->getAdminLink('AdminTawkto'),
+                'tab_id'     => (int)$this->context->controller->id,
+                'shops'      => $this->getStoreDetails()
```
Comment (GeekOfAges) [inline prestashop1.6/controllers/admin/AdminTawktoController.php:65]: Do not align on `=>` to maintain consistency

**Comment by GeekOfAges** [comment]
The concept here needs to be changed. I think here we should configure store that's selected in the main toolbar.
Then we can properly store the configuration entries for each shop/group and rely on built-in resolution to correctly maintain configuration hierarchy

**Pull request closed without merging**